src/components/Authentication/KratosUI.tsx

```diff
@@ -108,50 +161,72 @@ const KratosMessages: FC<{ messages?: Array<UiText> }> = ({ messages }) => {
 };
 
 const toUiControl = (node: UiNode, key: number) => {
-  switch (node.type) {
-    case 'input':
-      const attributes = node.attributes as UiNodeInputAttributes;
-      switch (attributes.type) {
-        case 'hidden':
-          return <KratosHidden key={key} node={node} />; // value={value} />;
-        case 'submit':
-          return <KratosButton key={key} node={node} />; //value={value} />;
-        case 'checkbox':
-          return <KratosCheckbox key={key} node={node} />; //value={value} />;
-        default:
-          return <KratosInput key={key} node={node} />; //value={value} />;
-      }
-    default:
-      return <KratosInput key={key} node={node} />; //value={value} />;
-  }
-};
-
-export const KratosUI: FC<KratosUIProps> = ({ flow }) => {
-  const nodes = flow?.ui.nodes || [];
-  const initialState: Partial<FormType> = {};
+  const attributes = node.attributes;
+  if (isUiNodeInputAttributes(attributes)) {
+    const variant = guessVariant(node);
 
-  nodes.forEach((node: UiNode) => {
-    const name = getNodeName(node);
-    const value = getNodeValue(node);
+    const extraProps: KratosInputExtraProps = {
+      autoCapitalize: 'off',
+      autoCorrect: 'off',
+    };
 
-    const key = name as keyof FormType;
-    initialState[key] = value || ('' as any);
-  });
+    switch (variant) {
+      case 'email':
+      case 'username':
+        extraProps.autoComplete = 'username';
+        break;
+      case 'password':
+        extraProps.autoComplete = 'password';
+        break;
+    }
 
-  console.log(initialState);
+    switch (attributes.type) {
+      case 'hidden':
+        return <KratosHidden key={key} node={node} />;
+      case 'submit':
+        return <KratosButton key={key} node={node} />;
+      case 'checkbox':
+        return <KratosCheckbox key={key} node={node} />;
+      default:
+        return <KratosInput key={key} node={node} {...extraProps} />;
+    }
+  } else {
+    return <KratosInput key={key} node={node} />;
+  }
+};
 
+export const KratosUI: FC<KratosUIProps> = ({ flow }) => {
   if (!flow) return null;
   const ui = flow.ui;
 
+  type NodeGroups = { default: []; oidc: []; password: [] };
+
+  const nodesByGroup: NodeGroups = flow.ui.nodes.reduce(
+    (acc: any, node: UiNode) => {
+      switch (node.group) {
+        case 'default':
+          return { ...acc, default: [...acc.default, node] };
+        case 'oidc':
+          return { ...acc, oidc: [...acc.oidc, node] };
+        case 'password':
+          return { ...acc, password: [...acc.password, node] };
+        default:
+          throw new Error(`Unknown node group ${node.group}`);
```

Comment:
How this error is handled? Unhandled exceptions will break the whole application! Also this one will be recalculated on every rerender. Could be memoized (`useMemo`).